test/support/pages/DiscoverPage.js

```diff
@@ -101,8 +107,30 @@ define(function (require) {
       .then(function (bars) {
         return bars;
       });
+    },
+
+    getChartInterval: function getChartInterval() {
+      return thisTime
+      .findByCssSelector('span.results-interval:nth-child(2) > a:nth-child(1)')
+      .getVisibleText();
+    },
+
+    setChartInterval: function setChartInterval(interval) {
+      return thisTime
+      .findByCssSelector('span.results-interval:nth-child(2) > a:nth-child(1)')
+      .click()
+      .catch(function () {
+        // in some cases we have the link above, but after we've made a
+        // selection we just have a select list.
+      })
+      .then(function () {
+        return thisTime
```

Comment:
I'm not familiar with using ```end``` but looking at the docs I don't think it would help in this case.  In the setChartInterval if there's a link we have to click it to get to the select list.  And then (regardless of the link) we have to find and click the option we want in the select list.  ```end``` seems to exit out of one level of scope.  I'm probably missing it but I don't see how I would use that in this case.